### PR TITLE
Add Gherkin scenarios for Android app context

### PR DESCRIPTION
**Pull request by radinamatic**
### Summary
Added Gherkin scenarios for signing in, facility and device settings relative to app context.
Run `pre-commit run --all-files`, green across the board.
Replaces #7023

### Reviewer guidance
Do the steps describe user flows as designed?
Is there anything missing?

### References
[Notion doc](https://www.notion.so/learningequality/Gherkin-stories-7b46e739971b4b92b2407dca8a7252ec)

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: integration_testing/features/app-context/app-context-disable-device-network-access.feature [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+Background:
+	Given that the device setting *Allow others in the network to access Kolibri on this device using a browser* is unchecked
+		And I use an external device to connect to the Kolibri app server and access Kolibri
+
+Scenario: Attempt to connect to the Kolibri app server from an external devices
+  When I type the Kolibri app server URL address on my external device browser
+  Then I see this message on the Kolibri sign in page: *Access to Kolibri has been restricted for external devices. To change this, sign in as a super admin and enable 'Allow others in the network to access Kolibri on this device using a browser', located in Device settings*
+
+Scenario: Security alert when network access setting is enabled
+	When I enable the device setting *Allow others in the network to access Kolibri on this device using a browser*
```
Comment (jredrejo) [inline integration_testing/features/app-context/app-context-disable-device-network-access.feature:15]: All this setting is not visible if you are not in an app context
In the last design change this was always visible, not when the setting is enabled. If that's how it should be, please fill an issue for it to be changed. Currently the security alert is always visible.

File: integration_testing/features/app-context/app-context-disable-device-network-access.feature [hunk 2]
```diff
@@ -0,0 +1,16 @@
+Feature: Disabled network access to Kolibri app server
+	If the network access to Kolibri server app is disabled, users who navigate to its URL need to see the notification message
+
+Background:
+	Given that the device setting *Allow others in the network to access Kolibri on this device using a browser* is unchecked
+		And I use an external device to connect to the Kolibri app server and access Kolibri
```
Comment (jredrejo) [inline integration_testing/features/app-context/app-context-disable-device-network-access.feature:6]: You don't need to be in an external device. An incongnito window in the browser or any other browser where the cookie for the App user is not set should trigger the error. Maybe asking to use an incognito window makes this test easier to be implemented.

File: integration_testing/features/app-context/app-context-remote-network-access.feature [hunk 3]
```diff
@@ -0,0 +1,24 @@
+Feature: Access Kolibri app server remotely
+	Users accessing Kolibri app server remotely need to have the same sign in experience
+
+	Background:
+		Given that the remote access to Kolibri is enabled
+
+	Scenario: Sign in directly after entering the username
+		Given that the signing in without password is enabled in *Facility > Settings*
+		When I try to access the Kolibri server app from a browser on an external device
```
Comment (jredrejo) [inline integration_testing/features/app-context/app-context-remote-network-access.feature:9]: or a different browser, or the same browser in an incognito window

**Review by jredrejo — COMMENTED** [review]
Leaved some comments that might require clarification

File: integration_testing/features/app-context/app-context-disable-device-network-access.feature (hunk 1 above)
Comment (radinamatic) [reply integration_testing/features/app-context/app-context-disable-device-network-access.feature:15]: > All this setting is not visible if you are not in an app context

This scenario is app-specific (goes to the separate folder), so if I understand correctly it applies.

> In the last design change this was always visible, not when the setting is enabled. If that's how it should be, please fill an issue for it to be changed. Currently the security alert is always visible.

Are the changes in #7051 meant to implement this new _visibility_?

File: integration_testing/features/app-context/app-context-disable-device-network-access.feature (hunk 2 above)
Comment (radinamatic) [reply integration_testing/features/app-context/app-context-disable-device-network-access.feature:6]: Good point, will add a comment bellow!

**Pull request merged**